minkindr/include/kindr/minimal/rotation-quaternion.h

```diff
@@ -156,9 +156,18 @@ class RotationQuaternionTemplate {
     return vector() == rhs.vector();
   }
 
+  /// \brief Check the validity of a rotation matrix.
+  static bool isValidRotationMatrix(const RotationMatrix& matrix);
+
+  /// \brief Factory to construct a RotationQuaternionTemplate from a near
+  ///        orthonormal rotation matrix.
+  inline static RotationQuaternionTemplate<Scalar> constructAndRenormalize(
+      const RotationMatrix& R) {
+    return RotationQuaternionTemplate<Scalar>(Implementation(R).normalized());
```

Comment:
A sorry, I did not look into the code following the link. This code looks like some thought in it :). Would be still interesting to compare it with a non approximative solution (in complexity and quality). On the other hand @kruesip did use Eigen conversions when he run into the problems that made him worry about it. But I don't recall which one. Probably not this one.